docs/topics/development/setup_and_configuration.md

```diff
@@ -0,0 +1,316 @@
+    - **[docker-compose.private.yml][docker-compose-private]**: Runs addons-server with the `customs` service that is only avaiable to Mozilla employees
+
+Our docker compose files rely on substituted values, all of which are included in our .env file for direct CLI compatibility.
+Any referenced `${VARIABLE}` in the docker-compose files will be replaced with the value from the .env file. We have tests
+that ensure any references are included in the .env file with valid values.
+
+This means when you run `make docker_compose_up`, the output on your machine will be exactly the same is if you ran
+`docker compose up  -d --wait --remove-orphans --force-recreate --quiet-pull` directly. You **should** use make commands,
+but sometimes you need to debug further what a command is running on the terminal and this architecture allows you to do that.
+
+By following these steps, you can set up your local development environment and understand the existing CI workflows for the **addons-server** project. For more details on specific commands and configurations, refer to the upcoming sections in this documentation.
+
+## Gotchas
+
+Here's a list of a few of the issues you might face when setting up your development environment
+
+### Can't access the web server?
+
+Check you've created a hosts file entry pointing `olympia.test` to the relevant IP address.
+
+If containers are failing to start use `docker compose ps` to check their running status.
+
+Another way to find out what's wrong is to run `docker compose logs`.
+
+### Getting "Programming error [table] doesn't exist"?
+
+Make sure you've run the `make initialize_docker` step as detailed in the initial setup instructions.
+
+### ConnectionError during initialize (elasticsearch container fails to start)
+
+When running `make initialize_docker` without a working elasticsearch container, you'll get a ConnectionError. Check the logs with `docker compose logs`. If elasticsearch is complaining about `vm.max_map_count`, run this command on your computer or your docker-machine VM:
+
+```sh
+sudo sysctl -w vm.max_map_count=262144
+```
+
+This allows processes to allocate more [memory map areas](https://stackoverflow.com/a/11685165/4496684).
+
+### Connection to elasticsearch timed out (elasticsearch container exits with code 137)
+
+`docker compose up -d` brings up all containers, but running `make initialize_docker` causes the elasticsearch container to go down. Running `docker compose ps` shows `Exited (137)` against it.
+
+Update default settings in Docker Desktop - we suggest increasing RAM limit to at least 4 GB in the Resources/Advanced section and click on "Apply and Restart".
+
+### Port collisions (nginx container fails to start)
+
+If you're already running a service on port 80 or 8000 on your host machine, the `nginx` container will fail to start. This is because the `docker-compose.override.yml` file tells `nginx` to listen on port 80 and the web service to listen on port 8000 by default.
```

Comment:
Somewhat off topic I guess, but if a collision on port 80 is a common occurrence, why don't we default to another port?  (And are there no other changes needed outside of the docker-compose.yml?)

Reply:
No idea.. this is copy pasted from exiting gotchas. I don't want to include modifying those under this PRs scope.